src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+		backgroundColor: theme.palette.grey.lighter,
+		display: "flex",
+		overflow: "hidden",
+		cursor: "default",
+	},
+	stickyHeaderTable: {
+		tableLayout: "fixed",
+		width: props => "calc(100% - " + (props.scrolled || "0") + "px)",
+	},
+	stickyHeaderTableScroll: {
+		width: props => (props.scrolled || "0") + "px",
+		backgroundColor: theme.palette.grey.lighter,
+		display: props => (props.scrolled > 0 ? "block" : "none"),
+	},
+	tableHeader: {
+		borderTop: props =>
+			props.withoutTopBorder ? "none" : "1px solid " + theme.palette.grey.borders,
+		borderBottom: props =>
+			props.stickyHeader ? "none" : "1px solid " + theme.palette.grey.borders,
+		display: props => (props.stickyHeader ? "none" : "table-header-group"),
+		backgroundColor: theme.palette.grey.lighter,
+		overflow: "hidden",
+		cursor: "default",
+	},
+	tableBody: {
+		overflowX: "hidden",
+		overflowY: "auto",
+	},
+	tableRow: {
+		borderBottom: "1px solid " + theme.palette.grey.borders,
+		cursor: props => (props.onRowClick ? "pointer" : "default"),
+		backgroundColor: theme.palette.background.paper + "!important",
+		"&:hover": {
+			backgroundColor: props =>
+				props.onRowClick ? theme.palette.grey.lighter : theme.palette.background.paper,
+		},
+	},
+	tableCell: {
+		padding: "20px 16px",
+		whiteSpace: "nowrap",
+		overflow: "hidden",
+		textOverflow: "ellipsis",
+	},
+	tableCellSelect: {
+		padding: "17px 16px",
+		width: "50px",
+	},
+	headerCell: {
+		padding: "11px 16px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	headerCellSelect: {
+		padding: "8px 16px",
+		width: "50px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	rowSelectCheckbox: {
+		padding: 0,
 	},
 }));
 
-const Table = ({ headers, rows, checkbox, placeholder }) => {
-	const classes = useStyles();
+function propsAreEqualRow(prev, next) {
+	return prev.selected === next.selected;
+}
+
+export const MemoTableRow = React.memo(TableRow, propsAreEqualRow);
+
+function propsAreEqualBody(prev, next) {
+	return (
+		next.contentTimestamp !== undefined &&
+		prev.contentTimestamp === next.contentTimestamp &&
+		prev.selectedNumber === next.selectedNumber
+	);
+}
+
+const TableBody = props => {
+	return <tbody className={props.className}>{props.rows}</tbody>;
+};
+
+export const MemoTableBody = React.memo(TableBody, propsAreEqualBody);
+
+const TableCell = withDeferredTooltip(
+	React.forwardRef((props, ref) => {
+		return (
+			<td className={props.className} ref={ref} {...props}>
+				{props.value}
+			</td>
+		);
+	}),
+);
+
+// We will assume for now that cell will never change value so we want to avoid rerendering as much as possible
+export const MemoTableCell = React.memo(TableCell, () => true);
+
+const buildRowCheckbox = (classes, key, selectionHandlers) => {
+	const rowSelected = selectionHandlers.isSelected(key);
+
+	return (
+		<td className={classes.tableCellSelect} selected={rowSelected}>
+			<CheckboxMui
```

Comment:
Isn't better here to use created wrapper for checkbox? I guess checkbox inside table are the same as others in design.

Reply:
The only reason why I had created a wrapper for Checkbox in the first place was to encapsulate "FormControlLabel" in order to avoid that hassle everytime we need it. There would no adding value to use our wrapper as we don't need the label for this case.